### PR TITLE
tests: workq: add support for null name parameter

### PR DESCRIPTION
**Pull request by ghost**
Validate k_work_queue_start() API with null name config, this should
not affect the name of queue's thread.

Signed-off-by: Lixin Guo <lixinx.guo@intel.com>

### CODE REVIEW

**Review by enjiamai — APPROVED** [review]

**Review by peter-mitsis — APPROVED** [review]

File: tests/kernel/workq/work/src/main.c [hunk 1]
```diff
@@ -113,6 +113,14 @@ static inline int preempt_counter(void)
...
 }
 
+static K_THREAD_STACK_DEFINE(invalid_test_stack, STACK_SIZE);
+static struct k_work_q invalid_test_queue;
+static atomic_t invalid_test_ctr;
+static inline int invalid_test_counter(void)
+{
+	return atomic_get(&invalid_test_ctr);
+}
+
```
Comment (ceolin) [inline tests/kernel/workq/work/src/main.c:123]: Are you using it anywhere ?

File: tests/kernel/workq/work/src/main.c [hunk 2]
```diff
@@ -244,6 +252,21 @@ static void test_queue_start(void)
...
+	zassert_equal(invalid_test_queue.flags, 0, NULL);
+	k_work_queue_start(&invalid_test_queue, invalid_test_stack, STACK_SIZE,
+			    PREEMPT_PRIORITY, &cfg);
+	zassert_equal(invalid_test_queue.flags, K_WORK_QUEUE_STARTED, NULL);
+
+	if (IS_ENABLED(CONFIG_THREAD_NAME)) {
+		const char *tn = k_thread_name_get(&invalid_test_queue.thread);
+
+		printk("%s\n", tn);
+		zassert_true(tn != cfg.name, NULL);
```
Comment (ceolin) [inline tests/kernel/workq/work/src/main.c:265]: You don't need this `printk`. If you need to print something please use `TC_PRINT`.

**Review by ceolin — CHANGES_REQUESTED** [review]
There are things to change, print to remove, function and variable not used ...

File: tests/kernel/workq/work/src/main.c (hunk 2 above)
Comment (LixinGuoX) [reply tests/kernel/workq/work/src/main.c:265]: Sorry for that, I will remove that and update in latest change, thank for pointing it out.

**Comment by LixinGuoX** [comment]
@ceolin could you help to review this PR again?

**Comment by LixinGuoX** [comment]
@ceolin could you help to review this PR again?

File: tests/kernel/workq/work/src/main.c [hunk 3]
```diff
@@ -244,6 +247,20 @@ static void test_queue_start(void)
...
+	cfg.name = NULL;
+	zassert_equal(invalid_test_queue.flags, 0, NULL);
+	k_work_queue_start(&invalid_test_queue, invalid_test_stack, STACK_SIZE,
+			    PREEMPT_PRIORITY, &cfg);
+	zassert_equal(invalid_test_queue.flags, K_WORK_QUEUE_STARTED, NULL);
+
+	if (IS_ENABLED(CONFIG_THREAD_NAME)) {
+		const char *tn = k_thread_name_get(&invalid_test_queue.thread);
+
+		zassert_true(tn != cfg.name, NULL);
```
Comment (ceolin) [inline tests/kernel/workq/work/src/main.c:259]: This assert is not necessary because of the next one.

**Review by ceolin — APPROVED** [review]

**Pull request merged**